lib/route-recognizer.js

```diff
@@ -85,32 +85,41 @@ EpsilonSegment.prototype = {
   generate: function() { return ""; }
 };
 
-function parse(route, names, types) {
+function parse(route, names, specificity) {
   // normalize route as not starting with a "/". Recognition will
   // also normalize.
   if (route.charAt(0) === "/") { route = route.substr(1); }
 
   var segments = route.split("/"), results = [];
 
+  // We take advantage of the fact that route specificity
+  // mirrors the way we write numbers as strings. For more on
+  // how this works, refer to:
+  // https://github.com/tildeio/route-recognizer/pull/46
```

Comment:
Having the information immediately at hand is better. Write the paragraph. Also: "the way we write numbers as strings" who is "we"? Do you actually mean "the way numbers and strings have a numerical value in ASCII"? or "the way JavaScript typecasts strings into numbers for comparisons", or something else?